lib/user-data.js

```diff
@@ -0,0 +1,68 @@
+import { IdentityVerification } from './index';
+import htmlencode from 'htmlencode';
+
+export default class UserData {
+  constructor(settings) {
+    this.loggedOut = !settings.user_id && !settings.email;
+
+    if (!settings.app_id) {
+      throw new Error('You must provide an app_id in your Intercom settings');
+    }
+    if (!this.loggedOut && !settings.verificationSecret) {
+      throw new Error('You must provide your verification secret in your Intercom settings');
+    }
+
+    this.settings = settings;
+  }
+  json() {
+    const verificationSecret = this.getVerificationSecret();
```

Comment:
We might also want to wrap the secret/hash things in a conditional and skip them if `!this.loggedOut`

Reply:
Yeah, it seems like we should skip this work if `this.loggedOut || this.settings.user_hash`. Since `verificationSecret` and `identifier` are only used in `setUserHash` in `json`, and `getVerificationSecret/getIdentifier` are sibling methods, you could move those getters to be called inside `setUserHash` as well.

Reply:
Great point